state/endpoint_bindings.go

```diff
@@ -167,6 +172,31 @@ func createEndpointBindingsOp(st *State, key string, givenMap map[string]string,
 	}, nil
 }
 
+func ensureEndpointSpaceID(st *State, givenMap map[string]string) (map[string]string, error) {
```

Comment:
this file has changes which impact almost all of the other changes in this pr, please review carefully.